File: Frontend/src/modules/FaultLinked/domain/model/faultlinked.ts
import { validator } from "@/modules/shared/domain/errors"

export type FaultLinked  =  {
    id: number;
	caseID: number;
    linkedID: number;
    faultID: number;
    dateRegistered: Date;
    case?: any;
    linked?: any;
    fault?: any;
}

export type FaultLinkedRow  = FaultLinked & {}

export type FaultLinkedFilter = CriteriaFilter & {
    name: string
}

export const validateFaultLinked = (faultlinked: FaultLinked) => {    
    const dataValidation = [
        { validation: !faultlinked.linkedID, error: 'ENTITY.must_specify_name', tag: 'linkedID' },
        { validation: !faultlinked.faultID, error: 'ENTITY.must_specify_name', tag: 'faultID' },
    ];

    validator(dataValidation);
}

File: Backend/src/CaseControl/CaseControl.Domain/Entities/FaultLinked.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CaseControl.Domain.Entities
{
    public class FaultLinked
    {
        [Key]
        public int ID { get; set; }
        public int CaseID { get; set; }
        public int LinkedID { get; set; }
        public int FaultID { get; set; }

        [Required]
        public DateTime DateRegistered { get; set; }

        public Case? Case { get; set; }
        public Linked? Linked { get; set; }
        public Fault? Fault { get; set; }
    }
}
